dev/generate

```diff
@@ -3,8 +3,8 @@ set -e
 
 go generate ./...
 
-mockgen -package api github.com/xmtp/xmtp-node-go/pkg/proto/mls/api/v1 MlsApi_SubscribeGroupMessagesServer,MlsApi_SubscribeWelcomeMessagesServer > pkg/mls/api/v1/mock.gen.go
-mockgen -package mocks -source ./pkg/mlsvalidate/service.go MLSValidationService > pkg/mlsvalidate/mocks/mock.gen.go
+# Generate mocks
+mockery
```

Comment:
Steven was previously using mockgen to generate mocks. I decided to switch us back to the more standard `testify/mock` and `mockgen`. It's more widely used and more actively maintained. Otherwise very similar, other than some syntax changes.

This probably should have been a separate PR, but a lot of the things that were being mocked were also changing.